test/fixtures/tls-check-extra-ca-certificates.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+const tls = require('tls');
+const assert = require('assert');
+
+const defaultSet = new Set(tls.getCACertificates('default'));
+const extraSet = new Set(tls.getCACertificates('extra'));
```

Comment:
Just a nit but this feels a bit weird creating a `Set` from the array in each of these cases. Do we anticipate that the returned array may return duplicates? If it does, should we filter those out when we create the list? Should we warn users in the documentation that duplicates may occur?

Reply:
I think it's possible to have duplicates in system certificates (you might install them in multiple stores, I have run into this case on Windows before, where it doesn't seem uncommon to drop the same certificates both in current user and local computer). And I don't think we validate the extras do not have duplicates?

Reply:
ok, that's what I suspected. Let's make sure to at least mention in the documentation for this that duplicates may occur.

Reply:
For optimizations' sake maybe the deduplication is something that we should handle after reading in the certificates; although for this PR specifically, seems fine to use a Set in the JS layer to deduplicate when building the cached arrays for now, and then follow-up with native layer deduplications.

Reply:
I think it's fine for us not to deduplicate as long as it is documented, and the use of `Set` here to avoid it in the tests is fine.

Reply:
Okay I'll just leave a note in the documentation for now. Although I think even if we implement deduplication in the native land, the `Set` in the tests are still needed because they are used for the convenient `Set.prototype.intersection` method to calculate supersets ;)